Tasks/MysqlDeploymentOnMachineGroupV1/sql/MysqlClient.ts

```diff
@@ -0,0 +1,111 @@
+import { ISqlClient } from './ISqlClient';
+
+import { MysqlTaskParameter } from '../models/MysqlTaskParameter';
+import { Utility } from '../operations/MysqlUtiliy';
+import task = require("vsts-task-lib/task");
+var packageUtility = require('webdeployment-common/packageUtility.js');
+import Q = require('q');
+
+export class MysqlClient implements ISqlClient {
+    private _mysqlTaskParameter: MysqlTaskParameter;
+    private _hostName: string;
+    private _toolPath: string;
+
+    constructor(mysqlTaskParameter: MysqlTaskParameter, toolPath: string) {
+        if (!mysqlTaskParameter) {
+            throw new Error(task.loc("MysqlTaskParameterCannotBeEmpty"));
+        }
+        if (!mysqlTaskParameter.getServerName() ||typeof mysqlTaskParameter.getServerName().valueOf() !== 'string') {
+            throw new Error(task.loc("MysqlServerNameCannotBeEmpty"));
+        }
+        if (!toolPath ||typeof toolPath.valueOf() !== 'string') {
+            throw new Error(task.loc("ToolPathCannotBeNull"));
+        }
+
+        this._mysqlTaskParameter = mysqlTaskParameter;
+        this._hostName = mysqlTaskParameter.getServerName();
+        this._toolPath = toolPath;
+    }
+
+    /**
+     * Get the connection argument for mysql
+     */
+    private _getArgumentString(): string{
+        let argumentString = "-h" + this._hostName + " -u" + this._mysqlTaskParameter.getSqlUserName() + " -p" + this._mysqlTaskParameter.getSqlPassword();
```

Comment:
Check for password with special characters `~!@#$%^&*()_+-={}[]|\:;"'`

Reply:
What is default timeout values for connection timeout and query timeout in MySQL?

Reply:
Default value is 0. Which means no timeout by default unless not explicitly set by user in mysql options for client or in mysql server setting options.